[SDK-3068] Add Auth-Vue beta quickstart (#9989)
auth0/docs: articles/quickstart/spa/vuejs-beta/02-calling-an-api.md

  - client
  - api
contentType: tutorial
useCase: quickstart
---

<!-- markdownlint-disable MD002 MD041 -->

<%= include('../_includes/_calling_api_create_api') %>

## Configuring the plugin
To call an API, configure the plugin by setting the `audience` to the **API Identifier** of the API you want to call:

```ts
import { createAuth0 } from '@auth0/auth0-vue';

const app = createApp(App);

app.use(
  createAuth0({
    domain: '<AUTH0_DOMAIN>',
    client_id: '<AUTH0_CLIENT_ID>',
    redirect_uri: '<MY_CALLBACK_URL>',
    audience: '<AUTH0_AUDIENCE>'
  })
);

app.mount('#app');
```

## Retrieving an Access Token
With the plugin configured using an `audience`, Auth0 will return an Access Token that can be sent to your API.
You can retrieve an Access Token by using the `getAccessTokenSilently` function thats exposed by our SDK.

```html
<script>
  import { useAuth0 } from '@auth0/auth0-vue';

  export default {
    setup() {
      const { getAccessTokenSilently } = useAuth0();

      return {
        doSomethingWithToken: async () => {
          const token = await getAccessTokenSilently();
        }
      };
    }
  };
</script>
```

### Using the Options API

> In you are using the Options API, you can use the same `getAccessTokenSilently` method from the global `$auth0` property through your component's `this`.

Reviewer: Typo, also the end reads a bit awkwardly. Consider:

```suggestion
If you are using the Options API, you can use the same `getAccessTokenSilently` method from the global `$auth0` property through the `this` accessor.
```

```html
<script>
  export default {
    methods: {
      async doSomethingWithToken() {
        const token = await this.$auth0.getAccessTokenSilently();
      }
    }
  };
</script>
```

## Calling an API

> Once you have an access token, you need to ensure that the token gets added to the `Authorization` header of your request.

Reviewer: Use [the active voice](https://github.com/auth0/docs/blob/master/STYLEGUIDE.md#voice) here. Consider:

```suggestion
To call an API, include the token in the `Authorization` header of your request.
```

> Depending on how you are implementing HTTP calls in your Vue application, the implementation will be slightly different. However, here is an example that uses `fetch` with Vue's Composition API:

Reviewer: Let's make this more concise. Consider:

```suggestion
There are many ways to make HTTP calls with Vue. Here is an example using the `fetch` API with Vue's Composition API:
```
